src/App.vue

```diff
@@ -1,47 +1,207 @@
 <script setup lang="ts">
-import HelloWorld from './components/HelloWorld.vue'
-import TheWelcome from './components/TheWelcome.vue'
+import Filter from './components/Filter.vue';
+import Button from './components/Button.vue';
+import MovieCard from './components/MovieCard.vue';
+import WebsiteTitle from './components/WebsiteTitle.vue';
+import Footer from './layout/Footer.vue';
+import { ref } from "vue";
+
+interface ButtonToggleArgs {
+    type: string,
+    title: string,
+}
+
+const searchFilterButtons = ref([
+    {
+        type: "search",
+        title: "TITLE",
+        active: true,
+        buttonSide: "left"
+    },
+    {
+        type: "search",
+        title: "GENRE",
+        active: false,
+        buttonSide: "right"
+    }
+]);
+
+const sortFilterButtons = ref([
+{
+        type: "sort",
+        title: "RELEASE DATE",
+        active: true,
+        buttonSide: "left"
+    },
+    {
+        type: "sort",
+        title: "RATING",
+        active: false,
+        buttonSide: "right"
+    }
+]);
+
+const toggleButton = (arg: ButtonToggleArgs) => {
+    if(arg.type == "search"){
+
+        searchFilterButtons.value = searchFilterButtons.value.map((btn) => {
+            if(btn.title === arg.title){
+                btn.active = true;
+            }else{
+                btn.active = false;
+            }
+            return btn;
```

Comment:
You can use spread operator if you want to do this type of operation.
`searchFilterButtons.value = searchFilterButtons.value.map(btn => ({ ...btn, active: btn.title === arg.title }));
`